tsdb/db.go

```diff
@@ -805,31 +816,65 @@ func (db *DB) Compact() (err error) {
 		// consistently, we explicitly remove the last value
 		// from the block interval here.
 		head := NewRangeHead(db.head, mint, maxt-1)
-		if err := db.compactHead(head); err != nil {
+		t, err := db.compactHead(head)
+		if err != nil {
 			return err
 		}
+		if t > walTruncationTime {
+			walTruncationTime = t
+		}
+		headCompacted = true
+	}
+
+	if headCompacted {
```

Comment:
What do you think to add a code comment to explain why we truncate the WAL only at the end?